src/mds/Capability.h

```diff
@@ -132,11 +103,47 @@ class Capability {
     void dump(Formatter *f) const;
     static void generate_test_instances(list<revoke_info*>& ls);
   };
-private:
-  __u32 _pending, _issued;
-  list<revoke_info> _revokes;
 
-public:
+
+  const static unsigned STATE_STALE		= (1<<0);
+  const static unsigned STATE_NEW		= (1<<1);
+
+
+  Capability(CInode *i = NULL, uint64_t id = 0, client_t c = 0) :
+    client_follows(0), client_xattr_version(0),
+    client_inline_version(0),
+    last_rbytes(0), last_rsize(0),
+    item_session_caps(this), item_snaprealm_caps(this),
+    item_revoking_caps(this), item_client_revoking_caps(this),
+    inode(i), client(c),
+    cap_id(id),
+    _wanted(0), num_revoke_warnings(0),
+    _pending(0), _issued(0),
+    last_sent(0),
+    last_issue(0),
+    mseq(0),
+    suppress(0), state(0) {
+    g_num_cap++;
+    g_num_capa++;
+  }
+  Capability(const Capability& other);  // no copying
+  ~Capability() {
+    g_num_cap--;
+    g_num_caps++;
+  }
+
+
+  static void *operator new(size_t num_bytes) {
+    void *n = pool.malloc();
+    if (!n)
+      throw std::bad_alloc();
+    return n;
+  }
+  void operator delete(void *p) {
+    pool.free(p);
+  }
+  const Capability& operator=(const Capability& other);  // no copying
```

Comment:
Do the code style rules really require us to separate this and the copy constructor when disabling both? (Or does it perhaps only allow us to use better solutions?)